oauth-common/src/main/java/io/strimzi/kafka/oauth/common/SSLUtil.java

```diff
@@ -27,11 +29,28 @@ public static SSLSocketFactory createSSLFactory(String truststore, String passwo
         }
 
         KeyStore store;
-        try (FileInputStream is = new FileInputStream(truststore)) {
-            store = KeyStore.getInstance(type != null ? type : KeyStore.getDefaultType());
-            store.load(is, password.toCharArray());
-        } catch (Exception e) {
-            throw new RuntimeException("Failed to load truststore: " + truststore, e);
+
+        if ("PEM".equals(type)) {
+            try (FileInputStream is = new FileInputStream(truststore)) {
+                store = KeyStore.getInstance("PKCS12");
+                store.load(null, null);
+
+                while (is.available() > 0) {
```

Comment:
FileInputStream.available() is not meant to be used this way - it can return 0 at any time, depending on the filesystem ...
(https://stackoverflow.com/questions/2788869/may-the-fileinputstream-available-foolish-me)

If using the code example from [here](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/security/cert/CertificateFactory.html) then it's really important (as hinted in the last sentence before the example) to wrap the FileInputStream with a BufferedInputStream.

```
        if ("PEM".equals(type)) {
            try (BufferedInputStream is = new BufferedInputStream(new FileInputStream(truststore)) {
```

Reply:
I didn't used that example, but it is a good point. I updated the code.